sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -47,20 +49,39 @@ object ResolveHints {
    *
    * This rule must happen before common table expressions.
    */
-  class ResolveBroadcastHints(conf: SQLConf) extends Rule[LogicalPlan] {
+  class ResolveBroadcastHints(conf: SQLConf, catalog: SessionCatalog) extends Rule[LogicalPlan] {
```

Comment:
Accordingly, we can use String instead of SessionCatalog.
```scala
-  class ResolveBroadcastHints(conf: SQLConf, catalog: SessionCatalog) extends Rule[LogicalPlan] {
+  class ResolveBroadcastHints(conf: SQLConf, currentDatabase: String) extends Rule[LogicalPlan] {
```

Reply:
We can instead use `getCurrentDatabase: () => String`?

Reply:
Ya. Right, please ignore this. We need `catalog` to lookup `global_temp`, too.